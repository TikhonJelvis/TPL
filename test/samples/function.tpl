print (id 10)
print (id id id id id 10)

print (const 10 100)
print (const 10 (print "fail"))

ignore (print "fail")

f x := x >< "bar "
g x := x >< "baz"
print ((g @ f) "foo")
print ((f @ g) "foo")

f x := x
print (f # 1 + 2 * 3)

f x := if (x = 10) 1 else 10
print (((=) `on` f) 1 2)
print (((=) `on` f) 1 10)
print (((=) `on` f) 10 10)
