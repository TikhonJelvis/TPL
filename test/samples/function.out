10
10
10
10
foobar baz
foobazbar 
7
true
false
true
null
